src/goto-programs/show_goto_functions.cpp

```diff
@@ -62,14 +62,24 @@ void show_goto_functions(
                      << (fun.second.body_available() ? ""
                                                      : ", body not available")
                      << " */";
+        msg.status() << messaget::eom;
       }
-
-      msg.status() << messaget::eom;
     }
     else
     {
-      goto_functions.output(ns, msg.status());
-      msg.status() << messaget::eom;
+      auto &out = msg.status();
+      for(const auto &fun : goto_functions.function_map)
+      {
+        if(fun.second.body_available())
+        {
+          out << "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n";
```

Comment:
I don't think it's a good idea to have code duplication - couldn't this also be fixed within `goto_functionst::output`? Else, that method should be removed.

Reply:
Removing might be the best solution. Code that is _almost_ duplicated is the prone to cause trouble.

Reply:
Done